agentMET4FOF/agents.py

```diff
@@ -1162,15 +1172,26 @@ def __init__(self, ip_addr="127.0.0.1", port=3333, connect=False, log_filename="
         # handle instantiating the dashboard
         # if dashboard_modules is False, the dashboard will not be launched
         if dashboard_modules is not False:
-            from .dashboard.Dashboard import AgentDashboard
+            # Initialize common dashboard parameters for both types of dashboards
+            # corresponding to different backends.
+            dashboard_params = {
+                "dashboard_modules": dashboard_modules,
+                "dashboard_layouts": [Dashboard_agt_net] + dashboard_extensions,
+                "dashboard_update_interval": dashboard_update_interval,
+                "max_monitors": dashboard_max_monitors,
+                "ip_addr": ip_addr,
+                "port": dashboard_port,
+                "agentNetwork": self,
+            }
+            # Initialize dashboard process/thread.
             if self.backend == "osbrain":
-                self.dashboard_proc = Process(target=AgentDashboard, args=(
-                    dashboard_modules, [Dashboard_agt_net] + dashboard_extensions, dashboard_update_interval,
-                    dashboard_max_monitors, ip_addr, dashboard_port, self))
+                from .dashboard.Dashboard import AgentDashboardProcess
+
+                self.dashboard_proc = AgentDashboardProcess(**dashboard_params)
             elif self.backend == "mesa":
-                self.dashboard_proc = Thread(target=AgentDashboard, args=(
-                    dashboard_modules, [Dashboard_agt_net] + dashboard_extensions, dashboard_update_interval,
-                    dashboard_max_monitors, ip_addr, dashboard_port, self))
+                from .dashboard.Dashboard import AgentDashboardThread
+
+                self.dashboard_proc = AgentDashboardThread(**dashboard_params)
```

Comment:
This is meant to improve readability of the fact, that both Process and Thread are initialised with the same parameters.